osu.Game/Screens/Ranking/CollectionButton.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Extensions;
+using osu.Framework.Graphics.Cursor;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Graphics.UserInterface;
+using osu.Game.Beatmaps;
+using osu.Game.Graphics;
+using osu.Game.Graphics.UserInterface;
+using osuTK;
+
+namespace osu.Game.Screens.Ranking
+{
+    public partial class CollectionButton : GrayButton, IHasPopover
+    {
+        private readonly BeatmapInfo beatmapInfo;
+
+        public CollectionButton(BeatmapInfo beatmapInfo)
+            : base(FontAwesome.Solid.Book)
+        {
+            this.beatmapInfo = beatmapInfo;
+
+            Size = new Vector2(75, 30);
+
+            TooltipText = "collections";
+        }
+
+        [BackgroundDependencyLoader]
+        private void load(OsuColour colours)
+        {
+            Background.Colour = colours.Green;
```

Comment:
Maybe this should be green only if the beatmap difficulty is in a collection?

Reply:
I have it implemented in local, but I was wondering about the fade times and performance.

https://github.com/user-attachments/assets/c791cf7f-acc7-4cd7-ae01-e17f09f2071b

The video is using a 500ms duration (same as replay download button next to it) but I personally feel like it's not responsive enough and would rather change it to 300ms like [UIV2 rounded button](https://github.com/ppy/osu/blob/565107205c60f14488f1642011808d286d712c31/osu.Game/Graphics/UserInterfaceV2/RoundedButton.cs#L90)

300ms:

https://github.com/user-attachments/assets/a8660bcc-48bf-45a8-a2d4-d633a87a89aa


Also in terms of performance, should I be able to do this, or is this ill advised?
```cs
// Might have performance issues idk how the realm works
realm.Realm.RealmChanged += (_, _) => updateRealm();
```

```cs
private void updateRealm()
{
    current.Value = realm.Realm.All<BeatmapCollection>().AsEnumerable().Any(c => c.BeatmapMD5Hashes.Contains(beatmapInfo.MD5Hash));
}
```

Reply:
I'm not fluent in realm myself, so you'll have to wait for another dev.

Reply:
seems @ppy/team-client is needed here.

Reply:
No, you can't use `RealmChanged`. The way to do it is to use `RealmAccess.RegisterForNotifications()`, allowing you to register a callback. There's a few example usages around the place.